<?xml version="1.0"?>
<launch>
    <!-- Terminal Input -->
    <node pkg="f1tenth_modules" name="KeyInput" type="KeyInput" output="screen"/>

    <!-- Joy stick driver-->
    <arg name="dev" default="/dev/input/js0" />
    <node pkg="joy" name="joy" type="joy_node">
        <param name="dev" value="$(arg dev)" /> <!-- Customize this to match the location your joystick is plugged in on-->
        <param name="deadzone" value="0.2" />
        <param name="autorepeat_rate" value="40" />
        <param name="coalesce_interval" value="0.025" />
    </node>

    <node pkg="f1tenth_modules" name="JoyStick" type="JoyStick" output="screen"/>

    <!-- RPLidar-->
    <node pkg="rplidar_ros" type="rplidarNode" name="rplidar_node" />

    <!-- VESC-->
    <arg name="vesc_config" default="$(find f1tenth_hardware)/config/vesc.yaml" />
    <rosparam file="$(arg vesc_config)" command="load" />

    <node pkg="vesc_ackermann" type="ackermann_to_vesc_node" name="ackermann_to_vesc" >
      <remap from="ackermann_cmd" to="vesc_cmd" />
    </node>

    <node pkg="vesc_driver" type="vesc_driver_node" name="vesc_driver" />

    <!-- MUX-->
    <node pkg="f1tenth_modules" name="MuxNode" type="MuxNode" output="screen">
        <!-- ??rosparams??-->
    </node>


    <!-- Wallfollowing -->
    <arg name="wallfollowing" default="0"/>
    <node pkg="f1tenth_modules" name="WallFollowing" type="WallFollowing" output="screen" if="$(eval arg('wallfollowing') == 1)">
        <rosparam command="load" file="$(find f1tenth_modules)/params.yaml"/>
    </node>

    <!-- GapFollowing -->
    <arg name="gapfollowing" default="0"/>
    <node pkg="f1tenth_modules" name="GapFollowing" type="GapFollowing" output="screen" if="$(eval arg('gapfollowing') == 1)">
        <rosparam command="load" file="$(find f1tenth_modules)/params.yaml"/>
    </node>

    <!-- EmergencyBrake -->
    <node pkg="f1tenth_modules" name="SafetyNode" type="SafetyNode" output="screen">
        <rosparam command="load" file="$(find f1tenth_modules)/params.yaml"/>
    </node>

</launch>